presto-main/src/main/java/com/facebook/presto/operator/unnest/UnnestOperator.java

```diff
@@ -196,111 +199,118 @@ public Page getOutput()
             return null;
         }
 
-        PageBuilderStatus pageBuilderStatus = new PageBuilderStatus(MAX_BYTES_PER_PAGE);
-        prepareForNewOutput(pageBuilderStatus);
-
-        int outputRowCount = 0;
-
-        while (currentPosition < currentPage.getPositionCount()) {
-            outputRowCount += processCurrentPosition();
-            currentPosition++;
-
-            if (outputRowCount >= MAX_ROWS_PER_BLOCK || pageBuilderStatus.isFull()) {
-                break;
-            }
-        }
-
-        Block[] outputBlocks = buildOutputBlocks();
+        int positionCount = currentPage.getPositionCount();
+        int batchSize = calculateNextBatchSize();
+        Block[] outputBlocks = buildOutputBlocks(batchSize);
 
-        if (currentPosition == currentPage.getPositionCount()) {
+        if (currentPosition == positionCount) {
             currentPage = null;
             currentPosition = 0;
         }
 
         return new Page(outputBlocks);
     }
 
-    private int processCurrentPosition()
+    private static Unnester createUnnester(Type nestedType, boolean isLegacyUnnest)
```

Comment:
looks like this method just moved? would you put mechanical refactorings/moves into a separate commit to make it easier to see material changes?